check_test_suite.py

```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/env python3.9
```

Comment:
I was not too keen on this either. But the scripts need to explicitly specify the correct python version. Otherwise, they pick up the pre-installed python 3.5.
From what I could find during my searches, it is not advisable to uninstall the existing python3.5 as it seems to be tightly coupled to the Ubuntu installation. Making a symlink from python3.9 to python3 wouldn't really help much either.

As for the python version, we can be on this version for as long as the repos host the required packages. Officially, python3.9 is supposed to be supported till 2024/25.